Source/Platform/DefaultOpenTK/Backend/Graphics/NativeWindow.cs

```diff
@@ -155,24 +156,50 @@ public NativeWindow(GraphicsMode mode, WindowOptions options)
 					"window icon from the game's main executable '{0}'. This is " +
 					"uncritical, but still an error: {1}",
 					executablePath,
-					LogFormat.Exception(e));
+					LogFormat.Exception(e)
+				);
 			}
 
 			if (options.ScreenMode == ScreenMode.FullWindow)
+			{
 				this.internalWindow.WindowState = WindowState.Fullscreen;
+			}
 
-			DualityApp.WindowSize = new Point2(this.internalWindow.ClientSize.Width, this.internalWindow.ClientSize.Height);
+			UpdateWindowSizeInfo();
 
 			// Register events and input
 			this.HookIntoDuality();
 
 			// Determine OpenGL capabilities and log them
 			GraphicsBackend.LogOpenGLSpecs();
 		}
+
+		private static VSyncMode GetVsyncModeFromOptions(WindowOptions options)
```

Comment:
Extracting this code and putting it into its own method is a good thing 👍 

Since this method only needs a `RefreshMode` value and nothing else, I'd argue that it should accept only a `RefreshMode` as a parameter, rather than the entire options instance, with naming adjusted accordingly.